org.scala-ide.sdt.core.tests/src/org/scalaide/core/semantichighlighting/classifier/RegionParser.scala

```diff
@@ -2,10 +2,44 @@ package org.scalaide.core.semantichighlighting.classifier
 
 import org.eclipse.jface.text.IRegion
 import org.eclipse.jface.text.Region
+import scala.annotation.tailrec
 
 object RegionParser {
 
   /**
+   * This class represents a substring with an optional prefix and suffix.
+   */
+  case class EmbeddedSubstr(str: String, prefix: String = "", suffix: String = "") {
+    private[RegionParser] val searchString = prefix + str + suffix
+  }
+
+  object EmbeddedSubstr {
+    implicit def wrapAsEmbeddedSubstring(str: String) = EmbeddedSubstr(str)
+  }
+
+
+  /**
+   * Extracts the regions marked by the given substrings.
+   */
+  def substrRegions(test: String, substrs: EmbeddedSubstr*): Map[IRegion, EmbeddedSubstr] = {
```

Comment:
Please describe more clearly how this method is supposed to work. See `delimitedRegions` for an example.